src/backend/tools/tavily.py

```diff
@@ -12,26 +14,79 @@ class TavilyInternetSearch(BaseTool):
 
     def __init__(self):
         self.client = TavilyClient(api_key=self.tavily_api_key)
+        self.num_results = 6
 
     @classmethod
     def is_available(cls) -> bool:
         return cls.tavily_api_key is not None
 
     def call(self, parameters: dict, **kwargs: Any) -> List[Dict[str, Any]]:
         query = parameters.get("query", "")
-        content = self.client.search(query=query, search_depth="advanced")
+        result = self.client.search(
+            query=query, search_depth="advanced", include_raw_content=True
+        )
 
-        if "results" not in content:
+        if "results" not in result:
             return []
 
+        expanded = []
+        for result in result["results"]:
+            # Append original search result
+            expanded.append(result)
+
+            # Get other snippets
+            snippets = result["raw_content"].split("\n")
+            for snippet in snippets:
+                if result["content"] != snippet:
+                    if (
+                        len(snippet.split()) > 10
+                    ):  # Skip snippets with less than 10 words
+                        new_result = {
+                            "url": result["url"],
+                            "title": result["title"],
+                            "content": snippet.strip(),
+                        }
+                        expanded.append(new_result)
+
+        reranked_results = self.rerank_page_snippets(
+            query, expanded, model=kwargs.get("model_deployment")
+        )
+
         return [
-            {
-                "url": result["url"],
-                "text": result["content"],
-            }
-            for result in content["results"]
+            {"url": result["url"], "text": result["content"]}
+            for result in reranked_results
         ]
 
+    def rerank_page_snippets(
```

Comment:
@BeatrixCohere @lusmoura

is there a reason why we consolidate reranking results [here](https://github.com/cohere-ai/cohere-toolkit/blob/main/src/backend/chat/collate.py#L9) instead of localizing it in the tools themselves? I feel it would be reasonable to do the rerank in the tool, as in this example we need a bit more flexibility on how the rerank is implemented.

Reply:
IMO it's okay to have custom implementations for each tool. But if the tool uses the default chunk and rerank, implementing it with all tool results will lead to less duplicated code

Reply:
Why do we need a custom implementation in this case?

Reply:
in this special case we want to rerank -> filter out results from the same url (only want one result from a url) -> return only top 6 results. if we do not do these extra filterings we end up with a ton of references/citations.

Reply:
Goctha! Maybe we should add a flag to indicate these documents shouldn't be reranked again with all the tool results?

Reply:
Revisiting this now. I think its fine that it gets reranked in `custom_chat` since that rerank function there reranks all tool_results and not for the specific tool.